### ISSUE TITLE
[bug] `color-palette` not respected for components inside the shadow root when the host is a provider

### ISSUE THREAD

**Issue by adamjohnson** [RedHat-UX/red-hat-design-system#1773]
## Describe the bug

If you have a component that is a `colorContextProvider` and nest an RHDS component in the shadowDOM of that provider, the nested component does not receive the appropriate color context from the host.

I'm currently nesting `rh-button` in the shadowDOM and not seeing the appropriate color palette on `rh-button` when `color-palette="dark"` is applied to the host.

## Screenshots

![Screenshot of a dark palette card with a close button next to the CTA that should be lighter than it is](https://github.com/user-attachments/assets/0c9603e7-0cdb-4b82-9700-b31787f1355f)

## Steps to reproduce

1. Clone RHDS, install dependencies and type `npm run dev`.
2. Go to `elements/rh-card/rh-card.ts`
3. On line 8, add `import '@rhds/elements/rh-button/rh-button.js';`
4. After line 104, hit enter and add `<rh-button variant="close">Close</rh-button>` (or anywhere inside the `render()` method.
5. Go to `elements/rh-card/demo/rh-card.html` and add `color-palette="dark"` to the card.
6. View the demo in your browser at http://localhost:8000/elements/card/demo/

## Results

While the `rh-card` has a dark color context, the `rh-button` element still displays in the light color context.

## Expected behavior

The `rh-button` should inherit the color palette from the host and display light on light contexts and dark on dark contexts.

## Operating System and Browsers

All

**Comment by adamjohnson** [comment]
@zeroedin Also mentioned this bug in this [`rh-video-embed` comment](https://github.com/RedHat-UX/red-hat-design-system/pull/1514#pullrequestreview-2235970933).

**Comment by bennypowers** [comment]
I took the following steps:
1. checkout clefairy branch
2. npm ci
3. npm run dev
4. navigate to http://localhost:8000/elements/card/demo/
5. run this script in the console:

```js
const card = $('rh-card')
card.colorPalette = 'darkest'
card.shadowRoot.getElementById('container').append(card.querySelector('rh-cta'))
```

I then observed that the cta, now in the shadow root, has the proper colours

I do believe that this addresses the issue as presented. It could be that adjustments on a per-element basis need to be made in #1853, so let's address any that come up over there.

**Comment by adamjohnson** [comment]
I tested the following components in the `rh-card.html` demo with `color-palette="dark"` applied to rh-card:

  * Button (close variant)
  * Site Status
  * Switch
  * Tile

Each of these are showing their light theme when the rh-card provider is `color-palette="dark"`:

![A screenshot of an rh-card on a dark color palette with several other elements in the shadowdom showing their light color palette](https://github.com/user-attachments/assets/193be2b7-6c2a-48e7-ae18-40c21fb9a22e)

Now that #1853 is merged, I ran a `git pull` on `staging/clefairy` and followed the instructions numbers 1-4 above. I then used [this code](https://pastebin.com/SAS2a9qH) in `elements/rh-card/rh-card.ts`.

Are you saying that each of these components needs their dark color palette's updated to the new theming tokens?

**Comment by bennypowers** [comment]
Accepting the outstanding to-do items in elements like table and audio player, etc. All of the theming issues should now be accounted for on the Clefairy branch.

If that's not the case, then this issue should be reopened.

**Comment by bennypowers** [comment]
I made some fixes, then added all those elements to the rh-card demo, then ran this in console:

```js
const card = $('rh-card');
      card.colorPalette = 'dark';
for (const el of $$([
  'rh-button',
  'rh-cta',
  'rh-site-status',
  'rh-switch',
  'rh-tile'
].join()))
  card.shadowRoot.getElementById('body').append(el)
```

and i got this

![image](https://github.com/user-attachments/assets/9d212544-eb94-4309-b91d-7dca4ce1831f)

so I think we can close this again. Thanks!

**Comment by adamjohnson** [comment]
Ran a `git pull` on `staging/clefairy` and this issue persists.

We're adding these elements to two different places.

  * I'm adding them to the `shadow-root` via `rh-card.ts`. 
  * In your example, they're being added to the `rh-card.html` demo.

When we add them to `rh-card.html`, everything works as expected (`.on.dark` is applied).

But when we add them to the `shadow-root` of a component, the color palette thinks they are `.on.light` and they display the light versions—even if the provider has `color-palette="dark"`.

![Screenshot of a dark rh-card with nested components displaying differently in the lightdom vs shadowdom](https://github.com/user-attachments/assets/6e41b98b-3eb9-41ac-af74-5beae82fabe7)

**Comment by bennypowers** [comment]
confirmed here, thanks for clarifying @adamjohnson

**Comment by bennypowers** [comment]
ok so it's not that the context-request events for those initially-root shadow children are orphaned - they don't ever make it to the document body

i'll keep investigating

**Comment by bennypowers** [comment]
ok i hope this is solved once and for all in 0c32ae92
